src/components/switch/Switch.tsx

```diff
@@ -0,0 +1,89 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
+
+import SwitchBase from './Switch.base';
+import { DEFAULT_SWITCH_STATE } from './Switch.constants';
+import { PSwitch } from './Switch.props';
+
+const baseProperties = ({ state = DEFAULT_SWITCH_STATE }: PSwitch): PSwitch => ({
+    state,
+    className: 'Switch',
+});
+
+const getSwitchVariables = ({
+    theme: { palette, text },
+    state = DEFAULT_SWITCH_STATE,
+}: ThemedStyledProps<PSwitch, TTheme>): FlattenSimpleInterpolation => {
+    let borderColor = palette.primary.main;
+    let textColor = text.primary;
+
+    switch (state) {
+        case 'on':
+            borderColor = palette.primary.main;
+            break;
+        case 'off':
+            borderColor = text.secondary;
+            break;
+        case 'disabled':
+            borderColor = text.disabled;
+            textColor = text.disabled;
+            break;
+        default:
+            break;
+    }
+
+    return css`
+        --var-background-color: ${borderColor};
+        color: ${textColor};
+        --switch-checked-color: ${palette.primary.main};
+    `;
+};
+
+const Switch = styled(SwitchBase).attrs(baseProperties)<PSwitch>`
+    ${getSwitchVariables};
+
+    .Switch--checkbox {
+        height: 0;
+        width: 0;
+        visibility: hidden;
+    }
+
+    .Switch--background {
+        display: flex;
+        align-items: center;
+        justify-content: space-between;
+        cursor: pointer;
+        background: var(--var-background-color);
+        border-radius: 100px;
+        position: relative;
+        transition: background-color 0.2s;
```

Comment:
we should use predefined values for animation times. This is something we definitely need to discuss. There is a really good section about that in the material docs: [speed](https://material.io/design/motion/speed.html#easing).